code/part-one/tests.js

```diff
@@ -0,0 +1,129 @@
+const chai = require('chai');
+const expect = chai.expect;
+const { hash } = require('./utils/helpers.js');
+const { Block, Blockchain, Transaction } = require('./blockchain.js');
+
+
+// Test transaction data
+const testFromAddress = hash('testFromAddress');
+const testToAddress = hash('testToAddress');
+const testAmount = '10.00';
+const transaction = new Transaction(
+  testFromAddress,
+  testToAddress,
+  testAmount
+);
+
+// Test blockchain data
+const blockchain = new Blockchain();
+const genesisBlock = blockchain.chain[0];
+const newBlock = new Block();
+const miningAddress = hash('miningAddress');
```

Comment:
To follow best practices for unit testing, much if not all of these data instantiations should be done in `beforeEach` hooks. This way each test gets fresh data that hasn't been modified by a previous test. I could be talked into including it here for simplicity's sake, as long as we're sure it's not going to impact any tests we want to write later.

Reply:
I can put them in the beforeEach hooks. I had them there originally but felt like I had to reuse the same variables a couple times. I don't mind moving them if it creates a cleaner test environment.